src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1376,7 +1387,29 @@ bool AdaptDispatch::HorizontalTabSet()
 // True if handled successfully. False otherwise.
 bool AdaptDispatch::ForwardTab(const size_t numTabs)
 {
-    return _pConApi->PrivateForwardTab(numTabs);
+    CONSOLE_SCREEN_BUFFER_INFOEX csbiex = { 0 };
+    csbiex.cbSize = sizeof(CONSOLE_SCREEN_BUFFER_INFOEX);
+    bool success = _pConApi->GetConsoleScreenBufferInfoEx(csbiex);
+    if (success)
+    {
+        const auto width = csbiex.dwSize.X;
+        const auto row = csbiex.dwCursorPosition.Y;
+        auto column = csbiex.dwCursorPosition.X;
+        auto tabsPerformed = 0u;
+
+        _InitTabStopsForWidth(width);
+        while (column + 1 < width && tabsPerformed < numTabs)
+        {
+            column++;
+            if (til::at(_tabStopColumns, column))
+            {
+                tabsPerformed++;
+            }
+        }
```

Comment:
just for my own edification: `SPC SPC TAB`

If the tabstops set are:

```
0       8      10
|       |       |
```

then

```
0       8      10
|       |       |
XX<TAB>
  ^ cursor
```

will result in

```
0       8      10
|       |       |
XX        |
          ^ cursor
```

right? Where we'll overshoot the tab by the number of columns we were off the previous tab?

Reply:
No. The algorithm doesn't care where it started from. It just keeps incrementing the column number until it reaches a tab stop - the `til::at(_tabStopColumns, column)` test will tell us if any given column is a tab stop or not.

In case the `tabsPerformed++` is confusing the matter, you can think of that as just being a `break` when all you need is a single tab. But because we're potentially performing multiple tabs, we have to keep track of the count, and only break once we've repeated the operation _numTabs_ times.

Reply:
D’oh, of course we break out after we hit the number of tabs requested without further incrementing `column`. Thanks for the reality check :)